rust/core-lib/src/event_context.rs

```diff
@@ -89,19 +89,19 @@ impl<'a> EventContext<'a> {
     /// to the current text. This is common to most edits that just modify
     /// selection or viewport state.
     fn with_view<R, F>(&mut self, f: F) -> R
-        where F: FnOnce(&mut View, &Rope) -> R
+        where F: FnOnce(&mut View, &Rope, &Client) -> R
     {
         let editor = self.editor.borrow();
         let mut view = self.view.borrow_mut();
-        f(&mut view, editor.get_buffer())
+        f(&mut view, editor.get_buffer(), &self.client)
     }
 
     pub(crate) fn do_edit(&mut self, cmd: EditNotification) {
         use self::EventDomain as E;
         let event: EventDomain = cmd.into();
         match event {
             E::View(cmd) => {
-                    self.with_view(|view, text| view.do_edit(text, cmd));
+                    self.with_view(|view, text, client| view.do_edit(text, cmd, client));
```

Comment:
maybe instead of passing in the client here, we could track if the find status is 'dirty', and then update the client during the normal render cycle? In general we've managed to avoid having either the `Editor` or the `View` having access to the client directly, which I think makes the code easier to reason about; you know that the only messages being sent to core are those that are either explicitly in some `EventContext` fn, or else are part of rendering the view.

An alternative to this would be to move the find commands into `EventDomain::Special`, which are handled individually in the `EventContext`, instead of just being forwarded to the `Editor` or `View`. I think I prefer the former, though? Updating the find state feels like it fits well with other types of client updates.